pkg/autoscaler/autoscaler.go

```diff
@@ -199,24 +203,39 @@ func (agg *perPodAggregation) podWeight(now time.Time) float64 {
 // Autoscaler stores current state of an instance of an autoscaler
 type Autoscaler struct {
 	*DynamicConfig
-	key          string
-	target       float64
-	stats        map[statKey]Stat
-	statsMutex   sync.Mutex
-	panicking    bool
-	panicTime    *time.Time
-	maxPanicPods float64
-	reporter     StatsReporter
-	targetMutex  sync.RWMutex
+	key             string
+	namespace       string
+	revisionService string
+	endpointsLister corev1listers.EndpointsLister
+	target          float64
+	stats           map[statKey]Stat
+	statsMutex      sync.Mutex
+	readyPods       map[time.Time]float64
```

Comment:
Does it have to be float? Seems like count. If this simplifies metric reporting down the line, I guess it's fine. But if it's just because math.Max() deals with floats64 only, then it might be better to write custom max() function.